File: examples/benchmark.py
#!/usr/bin/env python3
"""
Simple benchmark script for memory measurement
Performs basic computational tasks to generate measurable memory usage
"""

def fibonacci(n):
    """Generate fibonacci sequence up to n"""
    sequence = [0, 1]
    while len(sequence) < n:
        sequence.append(sequence[-1] + sequence[-2])
    return sequence

def prime_sieve(limit):
    """Sieve of Eratosthenes to find primes up to limit"""
    sieve = [True] * (limit + 1)
    sieve[0] = sieve[1] = False
    
    for i in range(2, int(limit**0.5) + 1):
        if sieve[i]:
            for j in range(i*i, limit + 1, i):
                sieve[j] = False
    
    return [i for i, is_prime in enumerate(sieve) if is_prime]

def memory_allocator():
    """Allocate some data structures to generate memory usage"""
    # Create lists and dictionaries to use memory
    data = []
    for i in range(1000):
        data.append({'id': i, 'value': str(i * 42), 'data': list(range(10))})
    
    # Generate fibonacci numbers
    fib = fibonacci(50)
    
    # Find prime numbers
    primes = prime_sieve(1000)
    
    # Print some results
    print(f"Generated {len(data)} data entries")
    print(f"Fibonacci(50): {len(fib)} numbers, last value: {fib[-1]}")
    print(f"Primes up to 1000: {len(primes)} found")
    print(f"Largest prime: {primes[-1]}")
    
    return len(data) + len(fib) + len(primes)

if __name__ == "__main__":
    print("PetriBench Memory Benchmark")
    print("-" * 30)
    
    result = memory_allocator()
    
    print(f"Total operations: {result}")
    print("Benchmark completed successfully")

File: examples/Benchmark.java
import java.util.*;

/**
 * Simple benchmark script for memory measurement in Java
 * Performs basic computational tasks to generate measurable memory usage
 */
public class Benchmark {
    
    static class DataEntry {
        int id;
        String value;
        List<Integer> data;
        
        DataEntry(int id, String value, List<Integer> data) {
            this.id = id;
            this.value = value;
            this.data = data;
        }
    }
    
    // Generate fibonacci sequence up to n
    public static List<Long> fibonacci(int n) {
        List<Long> sequence = new ArrayList<>();
        sequence.add(0L);
        sequence.add(1L);
        
        while (sequence.size() < n) {
            int size = sequence.size();
            long next = sequence.get(size - 1) + sequence.get(size - 2);
            sequence.add(next);
        }
        
        return sequence;
    }
    
    // Sieve of Eratosthenes to find primes up to limit
    public static List<Integer> primeSieve(int limit) {
        boolean[] sieve = new boolean[limit + 1];
        Arrays.fill(sieve, true);
        sieve[0] = sieve[1] = false;
        
        for (int i = 2; i * i <= limit; i++) {
            if (sieve[i]) {
                for (int j = i * i; j <= limit; j += i) {
                    sieve[j] = false;
                }
            }
        }
        
        List<Integer> primes = new ArrayList<>();
        for (int i = 2; i <= limit; i++) {
            if (sieve[i]) {
                primes.add(i);
            }
        }
        
        return primes;
    }
    
    // Allocate data structures to generate memory usage
    public static int memoryAllocator() {
        // Create list of objects to use memory
        List<DataEntry> data = new ArrayList<>();
        for (int i = 0; i < 1000; i++) {
            List<Integer> entryData = new ArrayList<>();
            for (int j = 0; j < 10; j++) {
                entryData.add(j);
            }
            
            DataEntry entry = new DataEntry(i, String.valueOf(i * 42), entryData);
            data.add(entry);
        }
        
        // Generate fibonacci numbers
        List<Long> fib = fibonacci(50);
        
        // Find prime numbers
        List<Integer> primes = primeSieve(1000);
        
        // Print some results
        System.out.println("Generated " + data.size() + " data entries");
        System.out.println("Fibonacci(50): " + fib.size() + " numbers, last value: " + fib.get(fib.size() - 1));
        System.out.println("Primes up to 1000: " + primes.size() + " found");
        System.out.println("Largest prime: " + primes.get(primes.size() - 1));
        
        return data.size() + fib.size() + primes.size();
    }
    
    public static void main(String[] args) {
        System.out.println("PetriBench Memory Benchmark");
        System.out.println("------------------------------");
        
        int result = memoryAllocator();
        
        System.out.println("Total operations: " + result);
        System.out.println("Benchmark completed successfully");
    }
}